### DISCUSSION TITLE
Multiple lights and overlays in same area

### DISCUSSION THREAD

**Discussion by DeathByDisco** [ExperienceLovelace/ha-floorplan#488] [Q&A] [answered]
Hi, 
This is my first time poster and new to ha-floorplan.
I have a base background of my floorplan where all light are off. Depending on the state of lights i load the overlays for each room. 
I got all my lights working fine.
I have a room (hallway) with 3 different light sources that blend together. Since I want to display my floor plan as realistically as possible, I would like to load the correct overlay based on different conditions. I created 5 different variations of the hallway image with different light sources rendered. 

-Main light - Image1.png
-Staircase light - Image2.png 
-Wall light - Image3.png 
-Main light + Staircase light - Image4.png
-Main light + Staircase light + Wall light - Image5.png

If Main light + Staircase light is turned on i would like to load Image4.png 
If Main light + Staircase light + Wall light is turned on i would like to load Image5.png 

Is there a way to get such condition checked and working ? 

Thank you

**Comment by OzGav** [comment, marked as answer]
You might find it easier to use a HA Automation to generate the correct image filename based on all those options you have and store that in an input_text and then have floorplan monitor that entity and load the image when the state changes. You can do it in floorplan but you will need to use javascript and the hass object which ha-floorplan exposes

**Reply by DeathByDisco** [reply]
Thanks for the fast response. 
I’m new to HA and haven’t use text_input or anything like that. 
I will read the documentation and try to learn how to use it. 
Thanks for the hint.
